src/fw-logs/fw-logs-parser.cpp

```diff
@@ -1,90 +1,215 @@
+            return sizeof( extended_fw_log_binary ) - sizeof( extended->info ) + extended->total_params_size_bytes;
         }
 
-        fw_log_data fw_logs_parser::parse_fw_log(const fw_logs_binary_data* fw_log_msg) 
+        fw_logs_parser::structured_binary_data
+        fw_logs_parser::structure_common_data( const fw_logs_binary_data * fw_log_msg ) const
         {
-            fw_log_data log_data = fw_log_data();
+            structured_binary_data structured_data;
 
-            if (!fw_log_msg || fw_log_msg->logs_buffer.size() == 0)
-                return log_data;
+            auto * log_binary = reinterpret_cast< const fw_log_binary * >( fw_log_msg->logs_buffer.data() );
 
-            log_data = fill_log_data(fw_log_msg);
+            structured_data.severity = static_cast< uint32_t >( log_binary->severity );
+            structured_data.source_id = static_cast< uint32_t >( log_binary->source_id );
+            structured_data.file_id = static_cast< uint32_t >( log_binary->file_id );
+            structured_data.module_id = static_cast< uint32_t >( log_binary->module_id );
+            structured_data.event_id = static_cast< uint32_t >( log_binary->event_id );
+            structured_data.line = static_cast< uint32_t >( log_binary->line_id );
+            structured_data.sequence = static_cast< uint32_t >( log_binary->seq_id );
 
-            //message
-            fw_string_formatter reg_exp(_fw_logs_formating_options.get_enums());
-            fw_log_event log_event_data;
-            _fw_logs_formating_options.get_event_data(log_data._event_id, &log_event_data);
+            return structured_data;
+        }
 
-            uint32_t params[3] = { log_data._p1, log_data._p2, log_data._p3 };
-            reg_exp.generate_message(log_event_data.line, log_event_data.num_of_params, params, &log_data._message);
+        void fw_logs_parser::structure_timestamp( const fw_logs_binary_data * raw,
+                                                  structured_binary_data * structured ) const
+        {
+            const auto actual_struct = reinterpret_cast< const extended_fw_log_binary * >( raw->logs_buffer.data() );
+            structured->timestamp = actual_struct->soc_timestamp;
+        }
 
-            //file_name
-            _fw_logs_formating_options.get_file_name(log_data._file_id, &log_data._file_name);
+        void fw_logs_parser::structure_params( const fw_logs_binary_data * raw,
+                                               size_t num_of_params,
+                                               structured_binary_data * structured ) const
+        {
+            const auto actual_struct = reinterpret_cast< const extended_fw_log_binary * >( raw->logs_buffer.data() );
+            if( num_of_params != actual_struct->number_of_params )
+                throw librealsense::invalid_value_exception( rsutils::string::from() << "Expecting " << num_of_params
```

Comment:
If we throw here, will we raise an error log or something and continue to the next log?

Reply:
Yes, but that depends on the user implementation.
`parse_fw_log` is called (through API `rs2_parse_firmware_log`) for each log message.
Both viewer and `rs-fw-logger` loop over all queued logs, catch exceptions and move to next log.Other users may have other error handling.